File: tests/fixtures/sample.py
def build_mini_graph(nodes):
    graph = {}
    for node in nodes:
        graph[node["id"]] = node
    return graph


class GraphWalker:
    def __init__(self, graph):
        self.graph = graph

    def walk(self, start_id, depth=3):
        visited = set()
        queue = [(start_id, 0)]
        while queue:
            node_id, d = queue.pop(0)
            if node_id in visited or d > depth:
                continue
            visited.add(node_id)
            node = self.graph.get(node_id)
            if node and "neighbors" in node:
                for nid in node["neighbors"]:
                    queue.append((nid, d + 1))
        return visited


def lazy_build_context(query):
    terms = query.split(" ")
    return [t for t in terms if len(t) > 2]


MAX_RESULTS = 100
DEFAULT_DEPTH = 2

File: tests/fixtures/sample.js
function buildMiniGraph(nodes) {
  const graph = new Map();
  for (const node of nodes) {
    graph.set(node.id, node);
  }
  return graph;
}

class GraphWalker {
  constructor(graph) {
    this.graph = graph;
  }

  walk(startId, depth = 3) {
    const visited = new Set();
    const queue = [{ id: startId, depth: 0 }];
    while (queue.length > 0) {
      const { id, depth: d } = queue.shift();
      if (visited.has(id) || d > depth) continue;
      visited.add(id);
      const node = this.graph.get(id);
      if (node && node.neighbors) {
        for (const nid of node.neighbors) {
          queue.push({ id: nid, depth: d + 1 });
        }
      }
    }
    return visited;
  }
}

function lazyBuildContext(query) {
  const terms = query.split(" ");
  return terms.filter(t => t.length > 2);
}

const MAX_RESULTS = 100;
const DEFAULT_DEPTH = 2;
